espflash/src/cli/config.rs

```diff
@@ -141,6 +141,26 @@ impl Config {
 
         let mut port_config = if let Ok(data) = read_to_string(&port_config_file) {
             toml::from_str(&data).into_diagnostic()?
+        } else if let Ok(data) = read_to_string(&project_config_file) {
+            if data.contains("[connection]") || data.contains("[[usb_device]]") {
```

Comment:
Once we decide that we need to do the migration, we also need to remove the keys from the project config, once the ports file has been succesfully saved. I'm pretty sure if we just call `Config::save_with` all this will be resolved right? It will write over the existing config file with the new format, plus write out the port info into the new file without any of this extra code?

Reply:
Hmm `Config::save_with` doesn't save the project config... but maybe it should?

Reply:
Ah okay, if it doesn't make sense to save the project config we can keep the special case save here then.